This patch, to be applied after `cert-file.patch', fixes compilation
on GNU/Hurd where `PATH_MAX' is not defined.

diff -ubB --show-c-function openssl-1.0.0e/crypto/x509/x509_def.c.orig openssl-1.0.0e/crypto/x509/x509_def.c
--- openssl-1.0.0e/crypto/x509/x509_def.c.orig	2012-01-06 00:08:48.000000000 +0100
+++ openssl-1.0.0e/crypto/x509/x509_def.c	2012-01-06 00:11:29.000000000 +0100
@@ -58,6 +58,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <limits.h>
 #include <unistd.h>
 #include <sys/types.h>
@@ -82,7 +83,7 @@ const char *X509_get_default_cert_dir(void)
 
 const char *X509_get_default_cert_file(void)
 {
-    static char buf[PATH_MAX] = X509_CERT_FILE;
+    static char *buf;
     static int init = 0;
     if (!init) {
         init = 1;
@@ -91,12 +92,14 @@ const char *X509_get_default_cert_file(void)
 #ifndef OPENSSL_SYS_WINDOWS
             if (getuid() == geteuid()) {
 #endif
-                strncpy(buf, s, sizeof(buf));
-                buf[sizeof(buf) - 1] = 0;
+                buf = strdup(s);
 #ifndef OPENSSL_SYS_WINDOWS
             }
 #endif
         }
+        else {
+          buf = strdup(X509_CERT_FILE);
+        }
     }
     return buf;
 }
